src/main/java/org/apache/commons/compress/archivers/sevenz/LZMA2Decoder.java

```diff
@@ -33,9 +34,12 @@
 
     @Override
     InputStream decode(final String archiveName, final InputStream in, final long uncompressedLength,
-            final Coder coder, final byte[] password) throws IOException {
+            final Coder coder, final byte[] password, int maxMemoryLimitInKb) throws IOException {
         try {
             final int dictionarySize = getDictionarySize(coder);
+            if ((dictionarySize / 1024) > maxMemoryLimitInKb) {
```

Comment:
I think you should call `LZMAInputStream.getMemoryUsage` rather than evaluate the dictionary size only. Too bad the constructor we use doesn't support the `memoryLimit` parameter other constructors allow.